internal/testcase/testcase_test.go

```diff
@@ -126,14 +116,25 @@ func TestNew(t *testing.T) {
 	}
 	for i, c := range cases {
 		tcs, err := New(bytes.NewReader([]byte(c.input)), "json")
-		if err != nil {
-			t.Errorf("Test %d: %q input: %s", i, c.input, err)
-			break
-		}
-		resultJSON := marshalTestCaseSet(t, tcs)
-		expectedJSON := marshalTestCaseSet(t, &c.expected)
-		if expectedJSON != resultJSON {
-			t.Errorf("Test %d:\nExpected:\n%s\nGot:\n%s", i, expectedJSON, resultJSON)
+		if c.expectedError != "" {
```

Comment:
Me personally, I don't like such conditions in the test code. In this case, I would consider to move these cases into a separate test function.

Reply:
I usually don't have a problem with it since the setup of the test often is significantly more code than the conditional itself, and splitting the test function in two duplicates all the setup code. Unless you extract the setup to a reusable function, but that has its own drawbacks.

However, in this particular case the split makes sense.